src/EventGrid/EventGrid/Domain/GetAzureEventGridDomain.cs

```diff
@@ -0,0 +1,190 @@
+        [Alias("DomainName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainResourceIdHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.ODataQueryHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ODataQuery { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = ResourceGroupNameParameterSet)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.TopHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateRange(1, 100)]
+        public int Top { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.NextLinkHelp,
+            ParameterSetName = NextLinkParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string NextLink { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string resourceGroupName = string.Empty;
+            string domainName = string.Empty;
+            IEnumerable<Domain> domainsList;
+            string newNextLink = null;
+            int? providedTop = null;
+
+            if (MyInvocation.BoundParameters.ContainsKey(nameof(this.Top)))
+            {
+                providedTop = this.Top;
+            }
+
+            if (!string.IsNullOrEmpty(this.ResourceId))
+            {
+                EventGridUtils.GetResourceGroupNameAndDomainName(this.ResourceId, out resourceGroupName, out domainName);
+            }
+            else if (!string.IsNullOrEmpty(this.Name))
```

Comment:
Thanks for the suggestion but unfortunatelly this does not build as I am getting the following error when using this.ResourceGroupName and this.Name as out parameters:

"A Property or indexer may not be passed as as an out or ref parameter'. Keeping it as is.

Reply:
As an FYI -- there's a [`ResourceIdentifier`](https://github.com/Azure/azure-powershell-common/blob/05d3ed69c8674f750faa8962e8314a83f02cc441/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs) type in our common code that you can use to break down the components of a resource id to get the resource group name, resource type, resource name, parent name (if applicable), etc. You can use that here to avoid the use of `out` variables and fix the consolidation issue. Let me know if this sounds like a reasonable fix.

Reply:
Thanks for the suggestion. This will work for the Domain Case but not the domain topic case as the domain topic case is a nest azure resource and not tracked azure resource. therefore, I changed this to use ResourceIdentifier in all Domain related cmdlets.